modules/extensions-support/src/main/java/org/mule/runtime/module/extension/internal/runtime/source/ExtensionMessageSource.java

```diff
@@ -621,9 +625,27 @@ protected void doInitialise() throws InitialisationException {
     flowConstruct = new LazyValue<>(() -> (FlowConstruct) componentLocator.find(getRootContainerLocation()).orElse(null));
     messageProcessContext = createProcessingContext();
     if (shouldRunOnThisNode()) {
+      if (LOGGER.isDebugEnabled()) {
+        boolean isPrimaryPollingInstance = clusterService.isPrimaryPollingInstance();
+        if (primaryNodeOnly) {
+          LOGGER
+              .debug("Message source '{}' on flow '{}' is being initialised, it is running on the primary node and it must only run on this node.",
+                     sourceModel.getName(), getLocation().getRootContainerName());
+        } else {
+          LOGGER
+              .debug("Message source '{}' on flow '{}' is being initialised, there are no cluster restrictions regarding which node must run the source. This node {} the primary node.",
+                     sourceModel.getName(), getLocation().getRootContainerName(), isPrimaryPollingInstance ? "is" : "is not");
+        }
+      }
+
       reallyDoInitialise();
     } else {
+      LOGGER
+          .debug("Message source '{}' on flow '{}' is not being initialised because it should only run on the primary node of the cluster.",
```

Comment:
Suggestions: 1. "Message source '{}' on flow '{}' is failed to initialize. Message source '{}' can only run on the primary node of the cluster."
2. Can we add details on which one is the primary node?

Reply:
Hi,

Regarding 1: it would be wrong to say it failed to initialized, it is not going to be initialized because of the cluster restriction applied in the source.
2 : Which kind of details were you thinking of? The only piece of information I see available is if the current node is the primary node or not.

Reply:
@ndinu 
#1. "Message source '{}' on flow '{}' cannot initialize. Message source '{}' can only run on the primary node of the cluster."
#2. Got your point. Let's leave it at "primary node" itself.